src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts

```diff
@@ -292,6 +289,7 @@ export class TableInteractorColumnComponent {
         }
       }
     }
+    return navigatorComponents;
```

Comment:
In this method, we are updating `this.navigatorComponents()` and also returning and setting the returned value to `navigatorComponentsTA()`, so `this.navigatorComponents()` and `navigatorComponentsTA()` are equals. We should treat `this.navigatorComponents()` as immutable, creating a new array and returning the new array.

Reply:
Also, are we not using `navigatorComponentsTA` in the HTML at all?

Reply:
I just pushed a change for renaming `navigatorComponentsTA` and also to use the new variable instead of `navigatorComponents`.